DESIGN.md

```diff
@@ -161,87 +161,88 @@ const (
 )
 
 const (
-    ContextSignature  = "Signature"
-    ContextSignature1 = "Signature1"
+    AlgorithmPS256   Algorithm = -37 // RSASSA-PSS w/ SHA-256 by RFC 8230
+    AlgorithmPS384   Algorithm = -38 // RSASSA-PSS w/ SHA-384 by RFC 8230
+    AlgorithmPS512   Algorithm = -39 // RSASSA-PSS w/ SHA-512 by RFC 8230
+    AlgorithmES256   Algorithm = -7  // ECDSA w/ SHA-256 by RFC 8152
+    AlgorithmES384   Algorithm = -35 // ECDSA w/ SHA-384 by RFC 8152
+    AlgorithmES512   Algorithm = -36 // ECDSA w/ SHA-512 by RFC 8152
+    AlgorithmEd25519 Algorithm = -8  // PureEdDSA by RFC 8152
 )
 
-var (
-    AlgorithmPS256   *Algorithm
-    AlgorithmPS384   *Algorithm
-    AlgorithmPS512   *Algorithm
-    AlgorithmES256   *Algorithm
-    AlgorithmES384   *Algorithm
-    AlgorithmES512   *Algorithm
-    AlgorithmEd25519 *Algorithm
-)
-
-type Algorithm struct {
-    Name     string
-    Value    int
-    HashFunc crypto.Hash
-}
+type Algorithm int64
+    func (a Algorithm) ComputeHash(data []byte) ([]byte, error)

```

Comment:
Is this function intended to be called by users, or just internally? If the later, we can leave it unexported.

Reply:
I've made it hidden :D